src/test/util/txmempool.cpp

```diff
@@ -116,3 +117,24 @@ std::optional<std::string> CheckPackageMempoolAcceptResult(const Package& txns,
     }
     return std::nullopt;
 }
+
+void CheckMempoolV3Invariants(const CTxMemPool& tx_pool, bool check_fees)
+{
+    LOCK(tx_pool.cs);
+    for (const auto& tx_info : tx_pool.infoAll()) {
+        const auto& entry = *Assert(tx_pool.GetEntry(tx_info.tx->GetHash()));
+        if (tx_info.tx->nVersion == 3) {
+            // Check that special v3 ancestor/descendant limits and rules are always respected
+            Assert(entry.GetCountWithDescendants() <= V3_DESCENDANT_LIMIT);
+            Assert(entry.GetCountWithAncestors() <= V3_ANCESTOR_LIMIT);
+            // If this transaction has at least 1 ancestor, it's a "child" and has restricted weight.
+            if (entry.GetCountWithAncestors() > 1) {
+                Assert(entry.GetTxSize() <= V3_CHILD_MAX_VSIZE);
+            }
+        }
+        // Transactions with fees of 0 or lower should be proactively trimmed.
```

Comment:
this only works if prioritization happens before mempool acceptance

Reply:
Yeah... slightly annoying that we can't check this until we're sure a trim happened.

Reply:
^We figured this out offline (see passed vs accepted), thanks